### DISCUSSION TITLE
Does not complete running inference, Please help

### DISCUSSION THREAD

**Discussion by king3052** [talmolab/sleap#2736] [Help!] [answered]
Hello everyone, hope you are having a good day,

I am running a tracking project on recorded single mice trials and finished mapping the plots. When I go to run the tracking, I get a run time error message below 

<img width="984" height="57" alt="image" src="https://github.com/user-attachments/assets/04ad2af6-98f9-4995-81b0-abd69a149126" />

It seems like it starts to train the module and then stops midway, how would I resolve this? Below I will attach the results to the sleap doctor and the settings for the tracking that I run. 

Thank you!!

<img width="1919" height="1031" alt="image" src="https://github.com/user-attachments/assets/0c5905b8-099e-4780-b8ed-25ba2c8c1a83" />
 
SLEAP System Diagnostics
========================
Generated: 2026-06-03 14:20:16

[Platform]
  OS:        Windows 11
  Platform:  Windows-11-10.0.26100-SP0
  Machine:   AMD64
  Processor: Intel64 Family 6 Model 198 Stepping 2, GenuineIntel
  Disk:      261.1 GB / 474.0 GB

[Python]
  Version:     3.13.13
  Executable:
C:\Users\Uteplab\AppData\Roaming\uv\tools\sleap\Scripts\python.exe
  Prefix:      C:\Users\Uteplab\AppData\Roaming\uv\tools\sleap

[UV]
  Version:         uv 0.11.15 (3cffe97c2 2026-05-18 x86_64-pc-windows-msvc)
  Path:            C:\Users\Uteplab\.local\bin\uv.EXE
  Cache Dir:       C:\Users\Uteplab\AppData\Local\uv\cache
  Tool Dir:        C:\Users\Uteplab\AppData\Roaming\uv\tools
  Tool Bin Dir:    C:\Users\Uteplab\.local\bin
  Python Dir:      C:\Users\Uteplab\AppData\Roaming\uv\python
  Installed Tools: sleap

[UV Config]
  Default Python:    (not configured)
  Resolved Python:
C:\Users\Uteplab\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation
.Python.3.13_qbz5n2kfra8p0\python.exe
  Python Preference: managed (default)
  Resolution:        highest (default)
  Index Strategy:    first-index (default)
  Prerelease:        if-necessary (default)

[GPU / CUDA]
  NVIDIA Driver: 595.71 (CUDA 13.2)
  GPU 0:         NVIDIA RTX 2000 Ada Generation (16118 MB free / 16380 MB)
  PyTorch:       v2.12.0+cu130 (CUDA 13.0)

[Packages]
  sleap:          v1.6.3 (pip)
  sleap-io:       v0.7.0 (pip)
  sleap-nn:       v0.2.0 (pip)
  numpy:          v2.4.6 (pip)
  h5py:           v3.16.0 (pip)
  PySide6:        v6.11.1 (pip)
  opencv-python:  v4.13.0.92 (pip)
  torch:          v2.12.0+cu130 (pip)
  imageio:        v2.37.3 (pip)
  imageio-ffmpeg: v0.6.0 (pip)
  av:             v15.1.0 (pip)

[CLI Binaries]
  sleap:
    Path:      C:\Users\Uteplab\.local\bin\sleap.EXE
    Real Path: C:\Users\Uteplab\.local\bin\sleap.exe
    Source:    pip
  sleap-nn-track:
    Path:      C:\Users\Uteplab\.local\bin\sleap-nn-track.EXE
    Real Path: C:\Users\Uteplab\.local\bin\sleap-nn-track.exe
    Source:    pip
  ffmpeg:
    Path:      C:\Users\Uteplab\scoop\shims\ffmpeg.EXE
    Real Path: C:\Users\Uteplab\scoop\shims\ffmpeg.exe
    Source:    PATH
    Python:    v8.0-full_build-www.gyan.dev
  ffmpeg (imageio):
    Path:
C:\Users\Uteplab\AppData\Roaming\uv\tools\sleap\Lib\site-packages\imageio_ffm
peg\binaries\ffmpeg-win-x86_64-v7.1.exe
    Source:    imageio-ffmpeg
    Python:    v7.1-essentials_build-www.gyan.dev

[PATH (relevant entries)]
  C:\Users\Uteplab\AppData\Roaming\uv\tools\sleap\Lib\site-packages\cv2\../..
/x64/vc17/bin
  C:\Users\Uteplab\.local\bin
  C:\Users\Uteplab\AppData\Local\Programs\Python\Launcher[/]

Copy this output when reporting issues at:
https://github.com/talmolab/sleap/issues

Tip: Use sleap doctor -o to save diagnostics to a file

PS C:\Users\Uteplab> sleap doctor

SLEAP System Diagnostics
========================
Generated: 2026-06-03 14:20:44

[Platform]
  OS:        Windows 11
  Platform:  Windows-11-10.0.26100-SP0
  Machine:   AMD64
  Processor: Intel64 Family 6 Model 198 Stepping 2, GenuineIntel
  Disk:      261.1 GB / 474.0 GB

[Python]
  Version:     3.13.13
  Executable:
C:\Users\Uteplab\AppData\Roaming\uv\tools\sleap\Scripts\python.exe
  Prefix:      C:\Users\Uteplab\AppData\Roaming\uv\tools\sleap

[UV]
  Version:         uv 0.11.15 (3cffe97c2 2026-05-18 x86_64-pc-windows-msvc)
  Path:            C:\Users\Uteplab\.local\bin\uv.EXE
  Cache Dir:       C:\Users\Uteplab\AppData\Local\uv\cache
  Tool Dir:        C:\Users\Uteplab\AppData\Roaming\uv\tools
  Tool Bin Dir:    C:\Users\Uteplab\.local\bin
  Python Dir:      C:\Users\Uteplab\AppData\Roaming\uv\python
  Installed Tools: sleap

[UV Config]
  Default Python:    (not configured)
  Resolved Python:
C:\Users\Uteplab\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation
.Python.3.13_qbz5n2kfra8p0\python.exe
  Python Preference: managed (default)
  Resolution:        highest (default)
  Index Strategy:    first-index (default)
  Prerelease:        if-necessary (default)

[GPU / CUDA]
  NVIDIA Driver: 595.71 (CUDA 13.2)
  GPU 0:         NVIDIA RTX 2000 Ada Generation (16118 MB free / 16380 MB)
  PyTorch:       v2.12.0+cu130 (CUDA 13.0)

[Packages]
  sleap:          v1.6.3 (pip)
  sleap-io:       v0.7.0 (pip)
  sleap-nn:       v0.2.0 (pip)
  numpy:          v2.4.6 (pip)
  h5py:           v3.16.0 (pip)
  PySide6:        v6.11.1 (pip)
  opencv-python:  v4.13.0.92 (pip)
  torch:          v2.12.0+cu130 (pip)
  imageio:        v2.37.3 (pip)
  imageio-ffmpeg: v0.6.0 (pip)
  av:             v15.1.0 (pip)

[CLI Binaries]
  sleap:
    Path:      C:\Users\Uteplab\.local\bin\sleap.EXE
    Real Path: C:\Users\Uteplab\.local\bin\sleap.exe
    Source:    pip
  sleap-nn-track:
    Path:      C:\Users\Uteplab\.local\bin\sleap-nn-track.EXE
    Real Path: C:\Users\Uteplab\.local\bin\sleap-nn-track.exe
    Source:    pip
  ffmpeg:
    Path:      C:\Users\Uteplab\scoop\shims\ffmpeg.EXE
    Real Path: C:\Users\Uteplab\scoop\shims\ffmpeg.exe
    Source:    PATH
    Python:    v8.0-full_build-www.gyan.dev
  ffmpeg (imageio):
    Path:
C:\Users\Uteplab\AppData\Roaming\uv\tools\sleap\Lib\site-packages\imageio_ffm
peg\binaries\ffmpeg-win-x86_64-v7.1.exe
    Source:    imageio-ffmpeg
    Python:    v7.1-essentials_build-www.gyan.dev

[PATH (relevant entries)]
  C:\Users\Uteplab\AppData\Roaming\uv\tools\sleap\Lib\site-packages\cv2\../..
/x64/vc17/bin
  C:\Users\Uteplab\.local\bin
  C:\Users\Uteplab\AppData\Local\Programs\Python\Launcher[/]

Copy this output when reporting issues at:
https://github.com/talmolab/sleap/issues

Tip: Use sleap doctor -o to save diagnostics to a file

**Comment by gitttt-1234** [comment, marked as answer]
Hi @king3052 !

That error is coming from your **single-instance** model, which assumes **exactly one instance per labeled frame**. The `14` vs `28` in the error  means at least one of your labeled frames has **2 instances** in it. The trainer tries to stack both animals' keypoints together, the shapes stop lining up, and training stops partway through.

To find those frames automatically, save the script below as `check_instances.py`, then run it (you already have `uv` installed):

```python
# /// script
# requires-python = ">=3.9"
# dependencies = ["sleap-io"]
# ///
import sys
import sleap_io as sio

labels = sio.load_slp(sys.argv[1])
bad = []
for lf in labels:
    n = len(lf.user_instances)
    if n > 1:
        print(f"{lf.video.filename}  frame {lf.frame_idx}: {n} instances")
        bad.append((lf.video.filename, lf.frame_idx, n))
print()
print(f"{len(bad)} frame(s) have more than one instance." if bad
      else "No frames with >1 instance found — likely a config/skeleton mismatch instead.")
```

Then run it from a terminal in that folder, pointing it at your project:

```
uv run check_instances.py "your_project.slp"
```

`uv` will grab the dependencies automatically and print every frame that has more than one animal. Open those frames in the GUI's **Labels** panel and delete the extra instance so each labeled frame has just a single mouse, then retrain it.

(If the script prints *no* offending frames, then it's most likely a different cause: were you reusing a training config or model from a project with a different skeleton? )

Let us know if that works!

UPDATE: Added a warning in the training pipeline https://github.com/talmolab/sleap-nn/pull/634


Thanks,

Divya

**Reply by king3052** [reply]
Hello Divya, 

Thank you for the prompt response. I will try it later in the day and let you know if it works!!
